pkg/openstack/loadbalancer.go

```diff
@@ -1150,11 +1150,15 @@ func (lbaas *LbaasV2) buildMonitorCreateOpts(svcConf *serviceConfig, port corev1
 	if port.Protocol == corev1.ProtocolUDP {
 		opts.Type = "UDP-CONNECT"
 	}
-	if svcConf.healthCheckNodePort > 0 && lbaas.canUseHTTPMonitor(port) {
-		opts.Type = "HTTP"
-		opts.URLPath = "/healthz"
-		opts.HTTPMethod = "GET"
-		opts.ExpectedCodes = "200"
+	if svcConf.healthCheckNodePort > 0 {
+		if lbaas.canUseHTTPMonitor(port) {
+			opts.Type = "HTTP"
+			opts.URLPath = "/healthz"
+			opts.HTTPMethod = "GET"
+			opts.ExpectedCodes = "200"
+		} else {
+			opts.Type = "TCP"
+		}
```

Comment:
> Do you think we could emit a warning event if we're doing a fallback to TCP?

That's why we output the type of health-monitor later on, so we can see it in the logs.
A warning might be more prominent, and I considered it, but we have not enough context in this function to output the affected load-balancer along with the warning.

Reply:
> > Also changing this that way breaks the UDP fallback. This unconditional `else` in line 1159 will invalidate `UDP-CONNECT` fallback.
> 
> This should not happen and I thought it could not. Looking at the code again, I'm not convinced any more either. Thanks for carefully looking! I'll come back with an updated patch, if needed, sorry.

The `healthCheckNodePort == 0` for `UDP`, so this won't break `UDP` AFAICS.

Reply:
No worries, this code is overly complicated, but so is caring about Octavia incompatibilities.

UDP can still have `healthCheckNodePort` if Service has ETP=Local. But I think with the check for `protocol=TCP` we're fine now.